community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/FreeIdKeeper.java

```diff
@@ -116,155 +148,165 @@ else if ( defraggedIdCount > 0 && canReadMoreIdBatches() )
 
     public long getCount()
     {
-        return defraggedIdCount;
+        return freeIdCount;
     }
 
     /*
-     * Returns true iff there are bytes between the current readPosition and maxReadPosition, i.e. there are more
-     * entries to read.
+     * After this method returns, if there were any entries found, they are placed in the readFromDisk list.
      */
-    private boolean canReadMoreIdBatches()
+    private boolean readIdBatch()
     {
-        assert (maxReadPosition - readPosition) % ID_ENTRY_SIZE == 0 :
-                String.format("maxReadPosition %d, readPosition %d do not contain an integral number of entries",
-                        maxReadPosition, readPosition);
-        return readPosition < maxReadPosition;
+        try
+        {
+            return readIdBatch0();
+        }
+        catch ( IOException e )
+        {
+            throw new UnderlyingStorageException( "Failed reading free id batch", e );
+        }
     }
 
-    /*
-     * After this method returns, if there were any entries found, they are placed in the readFromDisk list and the
-     * readPosition is updated accordingly.
-     */
-    private void readIdBatch()
+    private boolean readIdBatch0() throws IOException
```

Comment:
Not really a fun of `method0` patterns, maybe `methodImpl` instead?

Reply:
It's quite common.

Reply:
i know, it does force me to like it, he